Problem Statement - Food Nutrition

Intro to Python

## Data Dictionary
1. NDB_No - Nutrition database number
2. Shrt_Desc - Short description
3. Water_(g) - water in grams per 100 grams
4. Energ_Kcal - Energy in Kcal
5. Protein_(g) - Protein
6. LipidTot(g) - Total Lipid
7. Ash_(g) - Ash
8. Carbohydrt_(g) - Carbohydrate, by difference
9. FiberTD(g) - Fiber, total dietary
10. SugarTot(g) - Total Sugars
11. Calcium_(mg) - Calcium
12. Iron_(mg) - Iron
13. Magnesium_(mg) - Magnesium
14. Phosphorus_(mg) - Phosphorus
15. Potassium_(mg) - Potassium
16. Zinc_(mg) - Zinc
17. Copper_(mg) - Copper
18. Manganese_(mg) - Manganese
19. Selenium_(æg) - Selenium
20. Sodium_(mg) - Sodium

In [1]:
#Import all the necessary modules
#Import all the necessary modules
import pandas as pd
import numpy as np
import os

# Set the working directory

In [2]:
os.chdir('/Users/hemant/DataScience/Python/Week_2/MentorSession/')

# Import Excel file

Load the Data file into Python DataFrame using pandas read_excel method

The csv that we are going to load contains the unicode format data. We will use an additional encoding parameter which will allow pandas to load the csv. 

In [3]:
data_df = pd.read_excel('Food Nutrition.xlsx')

# View First 10 rows


 Head function is used to view the top records. The number of records to be viewed, needs to be  given in the parenthesis.

In [4]:
data_df.head(10)
#we can define how top rows we want to view

NDB_No               Shrt_Desc  Water_(g)  Energ_Kcal  Protein_(g)  \
0    1001        BUTTER,WITH SALT      15.87         717         0.85   
1    1002  BUTTER,WHIPPED,W/ SALT      16.72         718         0.49   
2    1003    BUTTER OIL,ANHYDROUS       0.24         876         0.28   
3    1004             CHEESE,BLUE      42.41         353        21.40   
4    1005            CHEESE,BRICK      41.11         371        23.24   
5    1006             CHEESE,BRIE      48.42         334        20.75   
6    1007        CHEESE,CAMEMBERT      51.80         300        19.80   
7    1008          CHEESE,CARAWAY      39.28         376        25.18   
8    1009          CHEESE,CHEDDAR      37.02         404        22.87   
9    1010         CHEESE,CHESHIRE      37.65         387        23.37   

   Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  \
0          81.11     2.11            0.06           0.0           0.06   
1          78.30     1.62            2.87           0.0           0.06   
2          99.48     0.00            0.00           0.0           0.00   
3          28.74      NaN            2.34           0.0           0.50   
4            NaN     3.18            2.79           0.0           0.51   
5          27.68     2.70            0.45           0.0           0.45   
6          24.26     3.68            0.46           0.0           0.46   
7          29.20     3.28            3.06           0.0           0.00   
8          33.31     3.71            3.09           0.0           0.48   
9          30.60     3.60            4.78           0.0           0.00   

   Calcium_(mg)  Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  \
0            24       0.02               2               24              24   
1            23       0.05               1               24              41   
2             4       0.00               0                3               5   
3           528       0.31              23              387             256   
4           674       0.43              24              451             136   
5           184       0.50              20              188             152   
6           388       0.33              20              347             187   
7           673       0.64              22              490              93   
8           710       0.14              27              455              76   
9           643       0.21              21              464              95   

   Sodium_(mg)  Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)  
0          643       0.09       0.000           0.000            1.0  
1          583       0.05       0.010           0.001            0.0  
2            2       0.01       0.001           0.000            0.0  
3         1146       2.66       0.040           0.009           14.5  
4          560       2.60       0.024           0.012           14.5  
5          629       2.38       0.019           0.034           14.5  
6          842       2.38       0.021           0.038           14.5  
7          690       2.94       0.024           0.021           14.5  
8          653       3.64       0.030           0.027           28.5  
9          700       2.79       0.042           0.012           14.5

# View Last 20 records

Tail function is used to view the last records. The number of records to be viewed, needs to be given in the parenthesis.

In [5]:
data_df.tail(20)

NDB_No                                          Shrt_Desc  Water_(g)  \
580    3934                   BABYFOOD,CORN & SWT POTATOES,STR      82.62   
581    3935  INF FORM, ABBO NUTR, SIMIL, ALIMENT, ADVAN, RT...      87.32   
582    3936  INF FORMULA,PBM PRODUC,STO BRA,RTF (FORMERLY W...      88.10   
583    3937  INF FORM,PBM PROD,STORE BRAND,LC,NOT REC (FORM...      76.00   
584    3938           INF FORMULA,PBM PRODUCTS,STORE BRAND,PDR       2.00   
585    3939       INF FORMULA,PBM PRODUCTS,STORE BRAND,SOY,RTF      88.00   
586    3940  INF FORMU,PBM PRODU,STORE BR,SOY,LIQ CONC,NOT ...      76.00   
587    3941       INF FORMULA,PBM PRODUCTS,STORE BRAND,SOY,PDR       2.00   
588    3942  INF FORMULA, MEAD JOHNS, ENFAMI, AR LIPIL, RTF...      87.00   
589    3943  INF FORMULA, MEA JOHNSO, ENFAMI, AR LIPIL, PDR...       3.00   
590    3944    INF FORM,ABBOT NUT,SIMIL NEOSU,RTF,W/ ARA & DHA      87.00   
591    3945  INF FORMULA, ABBOTT, SIMILAC, NEOSURE, PDR, W/...       2.25   
592    3946  INF FORMULA, ABB NUTR, SIMI,SENS (LACT FRE) RT...      87.00   
593    3947  INF FORMULA, ABB NUT,SIMIL,SENS,(LACT FR),LIQ ...      75.88   
594    3948  INF FORMULA, ABB NUTR,SIMIL,SENS,(LACTO FR), P...       2.20   
595    3949  INF FORMULA, ABBOTT NUTR, SIMILAC, ADVANCE, W/...      87.79   
596    3950  INF FORMULA,ABBOTT NUTR,SIMILAC,ADVANC,W/ IRON...       2.25   
597    3951  INF FORMU,ABBO NUTR,SIMILAC,ADVAN,W/ IRON,LIQ ...      76.16   
598    3952  INF FORMULA, ABB NUTR, SIMIL, ISOMIL, ADVA W/ ...      75.81   
599    3953  INF FORMULA, ABBO NUTR,SIMIL,ISOMIL, ADVANCE W...      87.54   

     Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  \
580          68         1.26           0.28     0.61           15.23   
581          67         1.80           3.63     0.48            6.77   
582          63         1.40           3.50     0.61            6.39   
583         130         2.90           7.00     0.61           13.90   
584         524        12.00          28.00     2.00           56.00   
585          63         1.80           3.50     0.61            6.09   
586         126         3.60           7.00     1.22           12.18   
587         508        13.60          27.20     5.00           52.20   
588          68         1.71           3.49     0.33            7.53   
589         509        12.73          25.97     2.46           56.03   
590          69         1.92           3.77     0.38            6.94   
591         520        14.42          28.33     3.26           51.75   
592          68         1.48           3.74     0.75            7.40   
593         128         2.73           6.95     0.80           13.64   
594         520        11.13          28.14     3.00           55.67   
595          66         1.36           3.70     0.38            6.77   
596         522        10.89          28.87     3.26           54.73   
597         127         2.64           6.89     0.79           13.52   
598         128         3.13           6.98     0.88           13.21   
599          66         1.61           3.59     0.66            6.70   

     Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  Iron_(mg)  Magnesium_(mg)  \
580           1.8           4.00            18       0.31              10   
581           0.0           4.40            69       1.18               5   
582           0.0           6.40            41       1.18               5   
583           0.0          13.90            83       2.36               9   
584           0.0          56.00           331       9.50              36   
585           0.0           6.10            59       1.18               7   
586           0.0          12.18           118       2.36              14   
587           0.0          52.20           453       9.10              51   
588           0.0           6.44            53       1.23               5   
589           0.0          56.03           397       9.16              41   
590           0.0          

Displaying the shape of the Data Frame in which first value is giving number of Rows and second value is giving number of columns.

In [6]:
data_df.shape

(600, 20)

In [7]:
data_df.size

12000

In [8]:
print("The number of rows are ",data_df.shape[0],"\n","The number of columns are",data_df.shape[1])

The number of rows are  600 
 The number of columns are 20


# Check for missing values

In [9]:
#Missing values can be - value at All, There might be Zero, NA or NAN
#Complete delete that row / imputation(Replacing - mean/median/linear regression/knn algo) 

data_df.isnull().sum()

NDB_No             0
Shrt_Desc          0
Water_(g)          0
Energ_Kcal         0
Protein_(g)        0
Lipid_Tot_(g)      1
Ash_(g)            1
Carbohydrt_(g)     0
Fiber_TD_(g)       0
Sugar_Tot_(g)      0
Calcium_(mg)       0
Iron_(mg)          0
Magnesium_(mg)     0
Phosphorus_(mg)    0
Potassium_(mg)     0
Sodium_(mg)        0
Zinc_(mg)          0
Copper_mg)         0
Manganese_(mg)     0
Selenium_(µg)      0
dtype: int64

In [10]:
data_df=data_df.dropna()
data_df

NDB_No                                          Shrt_Desc  Water_(g)  \
0      1001                                   BUTTER,WITH SALT      15.87   
1      1002                             BUTTER,WHIPPED,W/ SALT      16.72   
2      1003                               BUTTER OIL,ANHYDROUS       0.24   
5      1006                                        CHEESE,BRIE      48.42   
6      1007                                   CHEESE,CAMEMBERT      51.80   
..      ...                                                ...        ...   
595    3949  INF FORMULA, ABBOTT NUTR, SIMILAC, ADVANCE, W/...      87.79   
596    3950  INF FORMULA,ABBOTT NUTR,SIMILAC,ADVANC,W/ IRON...       2.25   
597    3951  INF FORMU,ABBO NUTR,SIMILAC,ADVAN,W/ IRON,LIQ ...      76.16   
598    3952  INF FORMULA, ABB NUTR, SIMIL, ISOMIL, ADVA W/ ...      75.81   
599    3953  INF FORMULA, ABBO NUTR,SIMIL,ISOMIL, ADVANCE W...      87.54   

     Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  \
0           717         0.85          81.11     2.11            0.06   
1           718         0.49          78.30     1.62            2.87   
2           876         0.28          99.48     0.00            0.00   
5           334        20.75          27.68     2.70            0.45   
6           300        19.80          24.26     3.68            0.46   
..          ...          ...            ...      ...             ...   
595          66         1.36           3.70     0.38            6.77   
596         522        10.89          28.87     3.26           54.73   
597         127         2.64           6.89     0.79           13.52   
598         128         3.13           6.98     0.88           13.21   
599          66         1.61           3.59     0.66            6.70   

     Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  Iron_(mg)  Magnesium_(mg)  \
0             0.0           0.06            24       0.02               2   
1             0.0           0.06            23       0.05               1   
2             0.0           0.00             4       0.00               0   
5             0.0           0.45           184       0.50              20   
6             0.0           0.46           388       0.33              20   
..            ...            ...           ...        ...             ...   
595           0.0           6.90            53       1.18               4   
596           0.0          54.73           410       9.47              32   
597           0.0          13.52            99       2.30               8   
598           0.0          13.10           134       2.30              10   
599           0.0           6.70            69       1.18               5   

     Phosphorus_(mg)  Potassium_(mg)  Sodium_(mg)  Zinc_(mg)  Copper_mg)  \
0                 24              24          643       0.09       0.000   
1                 24              41          583       0.05       0.010   
2                  3               5            2       0.01       0.001   
5                188             152          629       2.38       0.019   
6                347             187          842       2.38       0.021   
..               ...             ...          ...        ...         ...   
595               28              69           16       0.49       0.059   
596              221             552          126       3.94       0.473   
597               54             134           31       0.96       0.115   
598               96             138           56       0.96       0.096   
599               49              71           29       0.49       0.049   

     Manganese_(mg)  Selenium_(µg)  
0             0.000            1.0  
1             0.001            0.0  
2             0.000            0.0  
5             0.034           14.5  
6             0.038           14.5  
..              ...            ...  
595           0.003            1.2  
596           0.026           11.6  
597           0.006            2.8  
598    

There are two missing values in the data set one for each variable lipid_Tot_(g) and Ash_(g). We need to drop these missing values before proceeding. We will use the dropna method to drop the missing values.

In [11]:
data_df.isnull().sum()

NDB_No             0
Shrt_Desc          0
Water_(g)          0
Energ_Kcal         0
Protein_(g)        0
Lipid_Tot_(g)      0
Ash_(g)            0
Carbohydrt_(g)     0
Fiber_TD_(g)       0
Sugar_Tot_(g)      0
Calcium_(mg)       0
Iron_(mg)          0
Magnesium_(mg)     0
Phosphorus_(mg)    0
Potassium_(mg)     0
Sodium_(mg)        0
Zinc_(mg)          0
Copper_mg)         0
Manganese_(mg)     0
Selenium_(µg)      0
dtype: int64

In [12]:
data_df.shape

(598, 20)

In [13]:
data_df.size

11960

We can now see that there are no missing values in our data set.

# Summary of the data

To see the summary of any dataframe we will use describe function.

In [14]:
data_df.describe().T

count         mean          std     min        25%       50%  \
NDB_No           598.0  2305.448161  1081.807819  1001.0  1173.2500  2052.500   
Water_(g)        598.0    60.415870    31.929817     0.2    39.1675    76.180   
Energ_Kcal       598.0   186.521739   158.540534     0.0    63.0000   106.000   
Protein_(g)      598.0     8.640385    10.939719     0.0     1.7025     3.900   
Lipid_Tot_(g)    598.0     8.849649    12.262311     0.0     0.6975     3.475   
Ash_(g)          598.0     2.542040     6.494483     0.0     0.4900     0.890   
Carbohydrt_(g)   598.0    19.395920    21.522166     0.0     5.5600    10.840   
Fiber_TD_(g)     598.0     2.587458     7.594803     0.0     0.0000     0.000   
Sugar_Tot_(g)    598.0     9.122726    13.927320     0.0     0.5600     4.820   
Calcium_(mg)     598.0   262.015050   376.947430     0.0    23.2500   102.000   
Iron_(mg)        598.0     4.371622    12.162654     0.0     0.1825     0.415   
Magnesium_(mg)   598.0    41.760870    81.460504     0.0     8.0000    13.000   
Phosphorus_(mg)  598.0   188.610368   226.962501     0.0    35.0000   100.500   
Potassium_(mg)   598.0   311.349498   496.549282     0.0    95.0000   141.000   
Sodium_(mg)      598.0   348.270903  1830.690933     0.0    21.2500    50.000   
Zinc_(mg)        598.0     1.532475     1.970745     0.0     0.3000     0.670   
Copper_mg)       598.0     0.151381     0.279639     0.0     0.0260     0.050   
Manganese_(mg)   598.0     1.014246     4.220256     0.0     0.0210     0.140   
Selenium_(µg)    598.0    12.426254    21.833695     0.0     1.8000     3.650   

                       75%        max  
NDB_No           3190.7500   3953.000  
Water_(g)          85.9875     99.900  
Energ_Kcal        314.5000    876.000  
Protein_(g)        12.3850     84.080  
Lipid_Tot_(g)      13.0000     99.480  
Ash_(g)             3.2600     99.800  
Carbohydrt_(g)     19.1475     86.680  
Fiber_TD_(g)        1.2000     53.200  
Sugar_Tot_(g)      10.6925     74.460  
Calcium_(mg)      340.0000   2240.000  
Iron_(mg)           1.8975    123.600  
Magnesium_(mg)     33.0000    711.000  
Phosphorus_(mg)   264.2500   1349.000  
Potassium_(mg)    255.0000   4740.000  
Sodium_(mg)       170.7500  38758.000  
Zinc_(mg)           2.3600     17.460  
Copper_mg)          0.1100      2.467  
Manganese_(mg)      0.6770     60.127  
Selenium_(µg)      14.5000    208.100

# Create a vector “test” using the top 10 values of variable Protein_(g)

we will pass the variable name inside the square brackets of data_df and then use head function to retrieve the top 10 records for that variable.

In [15]:
test = data_df['Protein_(g)'].head(10)
test

0      0.85
1      0.49
2      0.28
5     20.75
6     19.80
7     25.18
8     22.87
9     23.37
10    23.76
11    11.12
Name: Protein_(g), dtype: float64

# Select the top 5 rows of initial 5 variables in a matrix format

To fetch the top 5 rows and initial variables we will use iloc and will pass the index both for rows and columns.

In [16]:
data_df.iloc[0:5,0:6]

NDB_No               Shrt_Desc  Water_(g)  Energ_Kcal  Protein_(g)  \
0    1001        BUTTER,WITH SALT      15.87         717         0.85   
1    1002  BUTTER,WHIPPED,W/ SALT      16.72         718         0.49   
2    1003    BUTTER OIL,ANHYDROUS       0.24         876         0.28   
5    1006             CHEESE,BRIE      48.42         334        20.75   
6    1007        CHEESE,CAMEMBERT      51.80         300        19.80   

   Lipid_Tot_(g)  
0          81.11  
1          78.30  
2          99.48  
5          27.68  
6          24.26

# What is the data_type of the Sodium_(mg) variable

To check the class of any variable we will use dtype property.

In [17]:
data_df['Shrt_Desc'].dtype

dtype('O')

In [18]:
data_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 598 entries, 0 to 599
Data columns (total 20 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   NDB_No           598 non-null    int64  
 1   Shrt_Desc        598 non-null    object 
 2   Water_(g)        598 non-null    float64
 3   Energ_Kcal       598 non-null    int64  
 4   Protein_(g)      598 non-null    float64
 5   Lipid_Tot_(g)    598 non-null    float64
 6   Ash_(g)          598 non-null    float64
 7   Carbohydrt_(g)   598 non-null    float64
 8   Fiber_TD_(g)     598 non-null    float64
 9   Sugar_Tot_(g)    598 non-null    float64
 10  Calcium_(mg)     598 non-null    int64  
 11  Iron_(mg)        598 non-null    float64
 12  Magnesium_(mg)   598 non-null    int64  
 13  Phosphorus_(mg)  598 non-null    int64  
 14  Potassium_(mg)   598 non-null    int64  
 15  Sodium_(mg)      598 non-null    int64  
 16  Zinc_(mg)        598 non-null    float64
 17  Copper_mg)      

# Create a new variable “EPW” by dividing Energ_Kcal with the Water; what is the dimension of the new dataset?


In [19]:
data_df['EPW'] = data_df['Energ_Kcal']/data_df['Water_(g)']

In [20]:
#dimension will be checked using the shape
data_df.shape

(598, 21)

In [21]:
data_df.head(5)

NDB_No               Shrt_Desc  Water_(g)  Energ_Kcal  Protein_(g)  \
0    1001        BUTTER,WITH SALT      15.87         717         0.85   
1    1002  BUTTER,WHIPPED,W/ SALT      16.72         718         0.49   
2    1003    BUTTER OIL,ANHYDROUS       0.24         876         0.28   
5    1006             CHEESE,BRIE      48.42         334        20.75   
6    1007        CHEESE,CAMEMBERT      51.80         300        19.80   

   Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  ...  \
0          81.11     2.11            0.06           0.0           0.06  ...   
1          78.30     1.62            2.87           0.0           0.06  ...   
2          99.48     0.00            0.00           0.0           0.00  ...   
5          27.68     2.70            0.45           0.0           0.45  ...   
6          24.26     3.68            0.46           0.0           0.46  ...   

   Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  Sodium_(mg)  \
0       0.02               2               24              24          643   
1       0.05               1               24              41          583   
2       0.00               0                3               5            2   
5       0.50              20              188             152          629   
6       0.33              20              347             187          842   

   Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)          EPW  
0       0.09       0.000           0.000            1.0    45.179584  
1       0.05       0.010           0.001            0.0    42.942584  
2       0.01       0.001           0.000            0.0  3650.000000  
5       2.38       0.019           0.034           14.5     6.897976  
6       2.38       0.021           0.038           14.5     5.791506  

[5 rows x 21 columns]

# Create a subset of the dataset, where the Energ_Kcal is less than 500, what is the dimension of this new dataset?


We will create a new data frame by using the subset conditions from the original data_df and will use shape property for checking the dimension.

In [22]:
data_df_new=data_df[data_df['Energ_Kcal']<500]

In [23]:
data_df_new.shape

(562, 21)

Here what we got reduced rows as there were 36 rows where Energ_Kcal > 500.
(598-36=562)

# Find the top 10 products based on following

1. Higher the Energy_Kcal, higher the ranking


Here we will use sorting to find out based on higher energy values.We will use sort_values method to order the data. Since we are saying higher trhe value the rank needs to be higher we will sort the data in the descending order so that the top most value gets the higher rank.

In [24]:
data_df.sort_values(by = 'Energ_Kcal' ,ascending=False).head(10)

NDB_No                             Shrt_Desc  Water_(g)  Energ_Kcal  \
2      1003                  BUTTER OIL,ANHYDROUS       0.24         876   
1      1002                BUTTER,WHIPPED,W/ SALT      16.72         718   
0      1001                      BUTTER,WITH SALT      15.87         717   
131    1145                   BUTTER,WITHOUT SALT      17.94         717   
125    1137                        EGG,YOLK,DRIED       3.08         669   
122    1134  EGG,WHL,DRIED,STABILIZED,GLUCOSE RED       1.87         615   
121    1133                         EGG,WHL,DRIED       2.78         592   
63     1070              DESSERT TOPPING,POWDERED       1.47         577   
178    1210                    EGG MIX,USDA CMDTY       2.78         555   
62     1069             CREAM SUBSTITUTE,POWDERED       2.98         529   

     Protein_(g)  Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  \
2           0.28          99.48     0.00            0.00           0.0   
1           0.49          78.30     1.62            2.87           0.0   
0           0.85          81.11     2.11            0.06           0.0   
131         0.85          81.11     0.09            0.06           0.0   
125        33.63          59.13     3.51            0.66           0.0   
122        48.17          43.95     3.63            2.38           0.0   
121        48.05          43.90     4.13            1.13           0.0   
63          4.90          39.92     1.17           52.54           0.0   
178        35.60          34.50     3.15           23.97           0.0   
62          2.48          32.92     2.34           59.29           0.0   

     Sugar_Tot_(g)  ...  Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  \
2             0.00  ...       0.00               0                3   
1             0.06  ...       0.05               1               24   
0             0.06  ...       0.02               2               24   
131           0.06  ...       0.02               2               24   
125           0.23  ...       9.56              26             1040   
122           0.00  ...       8.28              49              715   
121           0.56  ...       7.20              34              629   
63           52.54  ...       0.03               7               74   
178           2.46  ...       3.23              11              451   
62            7.77  ...       0.26               1              288   

     Potassium_(mg)  Sodium_(mg)  Zinc_(mg)  Copper_mg)  Manganese_(mg)  \
2                 5            2       0.01       0.001           0.000   
1                41          583       0.05       0.010           0.001   
0                24          643       0.09       0.000           0.000   
131              24           11       0.09       0.016           0.004   
125             264          149       7.73       0.000           0.185   
122             515          548       5.71       0.270           0.150   
121             540          476       3.15       0.203           0.058   
63              166          122       0.08       0.118           0.225   
178             373          576       2.76       0.149           0.056   
62              669          124       0.07       0.018           0.011   

     Selenium_(µg)          EPW  
2              0.0  3650.000000  
1              0.0    42.942584  
0              1.0    45.179584  
131            1.0    39.966555  
125          139.3   217.207792  
122          121.1   328.877005  
121          164.7   212.949640  
63             0.6   392.517007  
178          118.0   199.640288  
62             0.0   177.516779  

[10 rows x 21 columns]

2. Lower the water content, higher the ranking

Since we are saying lower the value the rank needs to be higher,we will sort the data in the ascending order so that the lower most value gets the higher rank.

In [25]:
data_df.sort_values(by='Water_(g)',ascending=False).head(10)

NDB_No                                     Shrt_Desc  Water_(g)  \
334    3024  BABYFOOD,H2O,BTLD,GERBER,WO/ ADDED FLUORIDE.      99.90   
301    2053                             VINEGAR,DISTILLED      94.78   
308    2068                              VINEGAR,RED WINE      94.47   
296    2048                                 VINEGAR,CIDER      93.81   
100    1112                               WHEY,ACID,FLUID      93.42   
102    1114                              WHEY,SWEET,FLUID      93.12   
380    3104                       BABYFOOD,VEG,SQUASH,STR      92.66   
381    3105                        BABYFOOD,VEG,SQUASH,JR      92.66   
375    3093                   BABYFOOD,GRN BNS,DICES,TODD      92.60   
374    3092                       BABYFOOD,VEG,GRN BNS,JR      92.50   

     Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  \
334           0         0.00           0.00     0.00            0.00   
301          18         0.00           0.00     0.02            0.04   
308          19         0.04           0.00     0.17            0.27   
296          21         0.00           0.00     0.17            0.93   
100          24         0.76           0.09     0.61            5.12   
102          27         0.85           0.36     0.53            5.14   
380          28         0.81           0.20     0.60            5.73   
381          24         0.81           0.20     0.60            5.73   
375          29         1.20           0.20     0.40            5.70   
374          24         1.20           0.10     0.40            5.80   

     Fiber_TD_(g)  Sugar_Tot_(g)  ...  Iron_(mg)  Magnesium_(mg)  \
334           0.0           0.00  ...       0.00               0   
301           0.0           0.04  ...       0.03               1   
308           0.0           0.00  ...       0.45               4   
296           0.0           0.40  ...       0.20               5   
100           0.0           5.12  ...       0.08              10   
102           0.0           5.14  ...       0.06               8   
380           0.9           3.37  ...       0.32              14   
381           0.9           3.24  ...       0.32              14   
375           1.3           1.06  ...       0.40              19   
374           1.9           1.08  ...       1.08              22   

     Phosphorus_(mg)  Potassium_(mg)  Sodium_(mg)  Zinc_(mg)  Copper_mg)  \
334                0               0            0       0.00       0.000   
301                4               2            2       0.01       0.006   
308                8              39            8       0.03       0.010   
296                8              73            5       0.04       0.008   
100               78             143           48       0.43       0.003   
102               46             161           54       0.13       0.004   
380               21             185            5       0.19       0.030   
381               21             185            5       0.19       0.030   
375               22             116           37       0.10       0.030   
374               19             128            5       0.19       0.049   

     Manganese_(mg)  Selenium_(µg)       EPW  
334           0.000            0.0  0.000000  
301           0.055            0.5  0.189913  
308           0.046           43.4  0.201122  
296           0.249            0.1  0.223857  
100           0.002            1.8  0.256904  
102           0.001            1.9  0.289948  
380           0.083            0.2  0.302180  
381           0.083            0.2  0.259011  
375           0.080            0.4  0.313175  
374           0.677            0.3  0.259459  

[10 rows x 21 columns]

# Create a subset of the data where product_desc contains “CHEESE” and list down the summary statistics of the subset


To find a specific value we will use the contains method. We will pass the string in the paranthesis which needs to be located in the dataframe variable.

In [26]:
data_df_cheese = data_df[data_df['Shrt_Desc'].str.contains('CHEESE')]

In [27]:
data_df_cheese.shape

(74, 21)

In [28]:
data_df_cheese.describe()

NDB_No  Water_(g)  Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  \
count    74.000000  74.000000   74.000000    74.000000      74.000000   
mean   1102.351351  48.021892  302.945946    20.956216      22.258378   
std      95.658702  14.181385   98.547797     6.423714       9.960468   
min    1006.000000  13.440000   72.000000     6.150000       0.000000   
25%    1024.250000  39.305000  264.000000    17.045000      19.480000   
50%    1042.500000  44.550000  330.000000    21.760000      25.320000   
75%    1181.750000  52.415000  372.500000    24.502500      29.800000   
max    1271.000000  82.480000  466.000000    37.860000      35.590000   

         Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  \
count  74.000000       74.000000     74.000000      74.000000     74.000000   
mean    4.102838        4.697568      0.005405       2.050541    632.459459   
std     1.730940        5.791562      0.028148       3.257006    296.165966   
min     1.020000        0.000000      0.000000       0.000000     53.000000   
25%     3.272500        2.010000      0.000000       0.322500    497.000000   
50%     3.790000        3.300000      0.000000       0.955000    676.500000   
75%     5.195000        5.395000      0.000000       2.365000    759.750000   
max     8.030000       42.650000      0.200000      23.670000   1375.000000   

       Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  \
count  74.000000       74.000000        74.000000       74.000000   
mean    0.433784       27.243243       472.648649      157.256757   
std     0.369797       14.965541       189.169666      169.498783   
min     0.000000        5.000000       107.000000       26.000000   
25%     0.180000       20.250000       377.500000       86.000000   
50%     0.370000       27.000000       465.500000      125.000000   
75%     0.582500       29.750000       580.750000      156.500000   
max     1.900000      115.000000       931.000000     1409.000000   

       Sodium_(mg)  Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)  \
count    74.000000  74.000000   74.000000       74.000000      74.000000   
mean    842.094595   2.609595    0.068986        0.081419      15.709459   
std     449.944263   1.094450    0.138540        0.181131       6.324104   
min      21.000000   0.330000    0.008000        0.002000       2.800000   
25%     601.000000   1.960000    0.025250        0.011000      14.500000   
50%     707.500000   2.800000    0.031500        0.022000      14.500000   
75%    1123.750000   3.500000    0.036000        0.038750      16.925000   
max    1809.000000   4.440000    0.732000        0.677000      35.800000   

             EPW  
count  74.000000  
mean    7.536921  
std     4.872384  
min     0.872939  
25%     4.894779  
50%     7.448191  
75%     9.510973  
max    34.672619

# Using the cut function on water variable divide the whole data into 6 bins, list down the summary statistics of all the 6 bins.  


In [29]:
data_bins = data_df
data_bins

NDB_No                                          Shrt_Desc  Water_(g)  \
0      1001                                   BUTTER,WITH SALT      15.87   
1      1002                             BUTTER,WHIPPED,W/ SALT      16.72   
2      1003                               BUTTER OIL,ANHYDROUS       0.24   
5      1006                                        CHEESE,BRIE      48.42   
6      1007                                   CHEESE,CAMEMBERT      51.80   
..      ...                                                ...        ...   
595    3949  INF FORMULA, ABBOTT NUTR, SIMILAC, ADVANCE, W/...      87.79   
596    3950  INF FORMULA,ABBOTT NUTR,SIMILAC,ADVANC,W/ IRON...       2.25   
597    3951  INF FORMU,ABBO NUTR,SIMILAC,ADVAN,W/ IRON,LIQ ...      76.16   
598    3952  INF FORMULA, ABB NUTR, SIMIL, ISOMIL, ADVA W/ ...      75.81   
599    3953  INF FORMULA, ABBO NUTR,SIMIL,ISOMIL, ADVANCE W...      87.54   

     Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  \
0           717         0.85          81.11     2.11            0.06   
1           718         0.49          78.30     1.62            2.87   
2           876         0.28          99.48     0.00            0.00   
5           334        20.75          27.68     2.70            0.45   
6           300        19.80          24.26     3.68            0.46   
..          ...          ...            ...      ...             ...   
595          66         1.36           3.70     0.38            6.77   
596         522        10.89          28.87     3.26           54.73   
597         127         2.64           6.89     0.79           13.52   
598         128         3.13           6.98     0.88           13.21   
599          66         1.61           3.59     0.66            6.70   

     Fiber_TD_(g)  Sugar_Tot_(g)  ...  Iron_(mg)  Magnesium_(mg)  \
0             0.0           0.06  ...       0.02               2   
1             0.0           0.06  ...       0.05               1   
2             0.0           0.00  ...       0.00               0   
5             0.0           0.45  ...       0.50              20   
6             0.0           0.46  ...       0.33              20   
..            ...            ...  ...        ...             ...   
595           0.0           6.90  ...       1.18               4   
596           0.0          54.73  ...       9.47              32   
597           0.0          13.52  ...       2.30               8   
598           0.0          13.10  ...       2.30              10   
599           0.0           6.70  ...       1.18               5   

     Phosphorus_(mg)  Potassium_(mg)  Sodium_(mg)  Zinc_(mg)  Copper_mg)  \
0                 24              24          643       0.09       0.000   
1                 24              41          583       0.05       0.010   
2                  3               5            2       0.01       0.001   
5                188             152          629       2.38       0.019   
6                347             187          842       2.38       0.021   
..               ...             ...          ...        ...         ...   
595               28              69           16       0.49       0.059   
596              221             552          126       3.94       0.473   
597               54             134           31       0.96       0.115   
598               96             138           56       0.96       0.096   
599               49              71           29       0.49       0.049   

     Manganese_(mg)  Selenium_(µg)          EPW  
0             0.000            1.0    45.179584  
1             0.001            0.0    42.942584  
2             0.000            0.0  3650.000000  
5             0.034           14.5     6.897976  
6             0.038           14.5     5.791506  
..              ...            ...          ...  
595           0.003            1.2     0.751794  
596           0.026           11.6   232.000000  
597           0.006            2.8 

In [30]:
data_bins['bins']=pd.cut(data_bins['Water_(g)'],6, labels = ['A','B','C','D','E','F'])

In [31]:
data_bins.head(10)

NDB_No                            Shrt_Desc  Water_(g)  Energ_Kcal  \
0     1001                     BUTTER,WITH SALT      15.87         717   
1     1002               BUTTER,WHIPPED,W/ SALT      16.72         718   
2     1003                 BUTTER OIL,ANHYDROUS       0.24         876   
5     1006                          CHEESE,BRIE      48.42         334   
6     1007                     CHEESE,CAMEMBERT      51.80         300   
7     1008                       CHEESE,CARAWAY      39.28         376   
8     1009                       CHEESE,CHEDDAR      37.02         404   
9     1010                      CHEESE,CHESHIRE      37.65         387   
10    1011                         CHEESE,COLBY      38.20         394   
11    1012  CHEESE,COTTAGE,CRMD,LRG OR SML CURD      79.79          98   

    Protein_(g)  Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  \
0          0.85          81.11     2.11            0.06           0.0   
1          0.49          78.30     1.62            2.87           0.0   
2          0.28          99.48     0.00            0.00           0.0   
5         20.75          27.68     2.70            0.45           0.0   
6         19.80          24.26     3.68            0.46           0.0   
7         25.18          29.20     3.28            3.06           0.0   
8         22.87          33.31     3.71            3.09           0.0   
9         23.37          30.60     3.60            4.78           0.0   
10        23.76          32.11     3.36            2.57           0.0   
11        11.12           4.30     1.41            3.38           0.0   

    Sugar_Tot_(g)  ...  Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  \
0            0.06  ...               2               24              24   
1            0.06  ...               1               24              41   
2            0.00  ...               0                3               5   
5            0.45  ...              20              188             152   
6            0.46  ...              20              347             187   
7            0.00  ...              22              490              93   
8            0.48  ...              27              455              76   
9            0.00  ...              21              464              95   
10           0.52  ...              26              457             127   
11           2.67  ...               8              159             104   

    Sodium_(mg)  Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)  \
0           643       0.09       0.000           0.000            1.0   
1           583       0.05       0.010           0.001            0.0   
2             2       0.01       0.001           0.000            0.0   
5           629       2.38       0.019           0.034           14.5   
6           842       2.38       0.021           0.038           14.5   
7           690       2.94       0.024           0.021           14.5   
8           653       3.64       0.030           0.027           28.5   
9           700       2.79       0.042           0.012           14.5   
10          604       3.07       0.042           0.012           14.5   
11          364       0.40       0.029           0.002            9.7   

            EPW  bins  
0     45.179584     A  
1     42.942584     A  
2   3650.000000     A  
5      6.897976     C  
6      5.791506     D  
7      9.572301     C  
8     10.913020     C  
9     10.278884     C  
10    10.314136     C  
11     1.228224     E  

[10 rows x 22 columns]

In [32]:
data_bins["bins"].value_counts()
#value_counts() : count unique values

F    193
E    175
A    127
C     52
D     41
B     10
Name: bins, dtype: int64

we can either do this in one go, like shown below

In [33]:
data_bins.groupby("bins").describe().T

bins                    A            B            C            D            E  \
NDB_No count   127.000000    10.000000    52.000000    41.000000   175.000000   
       mean   2427.000000  1291.200000  1127.615385  1225.097561  2379.885714   
       std    1006.346851   610.900938   103.775239   250.416234  1123.918049   
       min    1001.000000  1023.000000  1006.000000  1007.000000  1012.000000   
       25%    2003.500000  1034.250000  1029.750000  1072.000000  1183.000000   
...                   ...          ...          ...          ...          ...   
EPW    min       0.000000    10.971787     4.155412     1.551221     0.721154   
       25%      36.045815    12.462685     6.718986     2.846975     0.979906   
       50%      72.244898    14.511953     7.913460     4.373033     1.230272   
       75%     165.333333    17.114903     9.482712     5.405937     1.685755   
       max    3650.000000    39.966555    11.341632     6.651463     2.946455   

bins                    F  
NDB_No count   193.000000  
       mean   2757.367876  
       std     947.352355  
       min    1059.000000  
       25%    2053.000000  
...                   ...  
EPW    min       0.000000  
       25%       0.508475  
       50%       0.638468  
       75%       0.726846  
       max       1.163209  

[160 rows x 6 columns]

or we can create subsets of the data on the bases of bins

In [34]:
A=data_bins[data_bins["bins"]=="A"]
B=data_bins[data_bins["bins"]=="B"]
C=data_bins[data_bins["bins"]=="C"]
D=data_bins[data_bins["bins"]=="E"]
E=data_bins[data_bins["bins"]=="F"]
F=data_bins[data_bins["bins"]=="G"]

In [35]:
A.describe()# You can do for others
B.describe()
C.describe()
D.describe()
E.describe()
F.describe()

NDB_No  Water_(g)  Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  Ash_(g)  \
count     0.0        0.0         0.0          0.0            0.0      0.0   
mean      NaN        NaN         NaN          NaN            NaN      NaN   
std       NaN        NaN         NaN          NaN            NaN      NaN   
min       NaN        NaN         NaN          NaN            NaN      NaN   
25%       NaN        NaN         NaN          NaN            NaN      NaN   
50%       NaN        NaN         NaN          NaN            NaN      NaN   
75%       NaN        NaN         NaN          NaN            NaN      NaN   
max       NaN        NaN         NaN          NaN            NaN      NaN   

       Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  Iron_(mg)  \
count             0.0           0.0            0.0           0.0        0.0   
mean              NaN           NaN            NaN           NaN        NaN   
std               NaN           NaN            NaN           NaN        NaN   
min               NaN           NaN            NaN           NaN        NaN   
25%               NaN           NaN            NaN           NaN        NaN   
50%               NaN           NaN            NaN           NaN        NaN   
75%               NaN           NaN            NaN           NaN        NaN   
max               NaN           NaN            NaN           NaN        NaN   

       Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  Sodium_(mg)  \
count             0.0              0.0             0.0          0.0   
mean              NaN              NaN             NaN          NaN   
std               NaN              NaN             NaN          NaN   
min               NaN              NaN             NaN          NaN   
25%               NaN              NaN             NaN          NaN   
50%               NaN              NaN             NaN          NaN   
75%               NaN              NaN             NaN          NaN   
max               NaN              NaN             NaN          NaN   

       Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)  EPW  
count        0.0         0.0             0.0            0.0  0.0  
mean         NaN         NaN             NaN            NaN  NaN  
std          NaN         NaN             NaN            NaN  NaN  
min          NaN         NaN             NaN            NaN  NaN  
25%          NaN         NaN             NaN            NaN  NaN  
50%          NaN         NaN             NaN            NaN  NaN  
75%          NaN         NaN             NaN            NaN  NaN  
max          NaN         NaN             NaN            NaN  NaN

OR make loop for it.

In [36]:
X=[A,B,C,E,F,G]
for i in X:
    print(i.describe())

NameError: name 'G' is not defined